### DISCUSSION TITLE
Cancellation invoices

### DISCUSSION THREAD

**Discussion by tmechsner** [saleor/saleor#9655] [Q&A]
Hi community,
German law requires us to generate a cancellation invoice (actually it's a negative version of the original invoice) and a corrected invoice afterwards. I thought about writing a plugin that generates a cancellation invoice when "refund_payment" is called. However, generating a new, corrected invoice is not that easy, as the invoice plugin does not consider returned items. Does anyone else have this requirement and how did you solve it? Take the original order and iterate over all fulfillments filtered by status and accumulate (negative) quantities of fulfillment lines?

**Comment by karolkielecki** [comment]
Hi @tmechsner,
We don't recommend creating new plugins anymore. Instead, we recommend using apps. You can learn more about apps here: https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts. However, I can't guarantee if the app's approach would work in that case. We need to know more details--could you please tell us your specific requirements?

**Reply by tmechsner** [reply]
Hi @karolkielecki, thanks for your reply. I know about the preference of apps over plugins, but the latter is easier and faster to implement and can easily be deployed via CI/CD, which is not the case for apps - and our customer is very short on budget... 
The requirement is that we need to generate a cancellation invoice as soon as an item or a complete order is refunded. This cancellation invoice negates the original invoice. It needs to contain some special text and a reference to the original invoice id. Afterwards, a new invoice including the remaining order items needs to be generated. This is a requirement by German law.

**Reply by karolkielecki** [reply]
IMO in such a case, you will need to integrate (app) your e-commerce project with accounting software.

**Reply by tmechsner** [reply]
Okay, thanks for your reply.
